cmd/config/init/flags.go

```diff
@@ -67,24 +67,27 @@ func GetInitConfig(initCmd *cobra.Command, args []string) (config.RollappConfig,
 	cfg.TokenSupply = tokenSupply
 	cfg.DA = config.DAType(strings.ToLower(initCmd.Flag(FlagNames.DAType).Value.String()))
 	cfg.VMType = config.VMType(initCmd.Flag(FlagNames.VMType).Value.String())
-	return formatBaseCfg(cfg, initCmd), nil
+	return formatBaseCfg(cfg, initCmd)
 }
 
-func formatBaseCfg(cfg config.RollappConfig, initCmd *cobra.Command) config.RollappConfig {
+func formatBaseCfg(cfg config.RollappConfig, initCmd *cobra.Command) (*config.RollappConfig, error) {
 	setDecimals(initCmd, &cfg)
 	formattedRollappId, err := generateRollappId(cfg)
 	if err != nil {
-		return cfg
+		return nil, err
 	}
 	cfg.RollappID = formattedRollappId
-	return cfg
+	return &cfg, nil
 }
 
 func generateRollappId(rlpCfg config.RollappConfig) (string, error) {
 	for {
 		RandEthId := generateRandEthId()
 		isUnique, err := isEthIdentifierUnique(RandEthId, rlpCfg)
 		if err != nil {
+			if rlpCfg.HubData.ID == LocalHubID {
```

Comment:
move it before `isUnique, err := isEthIdentifierUnique(RandEthId, rlpCfg)`
no need to check if it's random with local hub